Doc/library/typing.rst

```diff
@@ -1500,6 +1510,70 @@ These are not used in annotations. They are building blocks for declaring types.
    ``True`` as the value of the ``total`` argument. ``True`` is the default,
    and makes all items defined in the class body required.
 
+   It is possible for a ``TypedDict`` type to inherit from one or more TypedDict types
+   using the class-based syntax.
+   Usage::
+
+      class Point3D(Point2D):
+         z: int
+
+   Now the ``Point3D`` has three items: ``x``, ``y`` and ``z``. It is equivalent to this
+   definition::
+
+      class Point3D(TypedDict):
+         x: int
+         y: int
+         z: int
+
+   The type info for introspection can be accessed via ``Point2D.__annotations__``,
+   ``Point2D.__total__``, ``Point2D.__required_keys__``, and
+   ``Point2D.__optional_keys__``.
+
+   .. attribute:: __annotations__
+
+      ``Point2D.__annotations__`` gives a dict maps the field names to the field types::
+
+         >>> from typing import TypedDict
+         >>> class Point2D(TypedDict):
+         ...     x: int
+         ...     y: int
+         ...
+         >>> Point2D.__annotations__
+         {'x': <class 'int'>, 'y': <class 'int'>}
+
+   .. attribute:: __total__
+
+      ``Point2D.__total__`` gives the value of the ``total`` argument::
+
+         >>> class Point2D(TypedDict): pass
+         >>> Point2D.__total__
+         True
+         >>> class Point2D(TypedDict, total=False): pass
+         >>> Point2D.__total__
+         False
+
+   .. attribute:: __required_keys__
+   .. attribute:: __optional_keys__
+
+      ``Point2D.__required_keys__`` and ``Point2D.__optional_keys__`` return
+      ``frozenset`` objects containing required and optional keys, respectively.
```

Comment:
PEP 655 (Required[] and NotRequired[] for TypedDict) deliberately uses the phrasing "potentially-missing keys" to refer to the concept described by `__optional_keys__` to avoid confusion with Optional[], which is an entirely different concept.

So:

* I might suggest replacing `optional keys` with `potentially-missing keys` here (and in similar locations).
* I could also see using the phrase `not-required keys` to have a different kind of alignment with the upcoming PEP 655.

FWIW, I think it's probably too late to change the name of `__optional_keys__` to avoid confusion with Optional[].